src/util/strencodings.cpp

```diff
@@ -579,13 +579,13 @@ std::string Capitalize(std::string str)
 
 std::string HexStr(const Span<const uint8_t> s)
 {
-    std::string rv;
+    std::string rv(s.size() * 2, '\0');
     static constexpr char hexmap[16] = { '0', '1', '2', '3', '4', '5', '6', '7',
                                          '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
-    rv.reserve(s.size() * 2);
+    auto* out = rv.data();
```

Comment:
Is it allowed to write to a `std::string`'s backing memory? I vaguely remember it was not, but maybe that was pre C++11.

Reply:
Before C++11 it was not guaranteed by the standard that std::string's memory is contiguous, but as far as I know all implementations did that anyways. It is certainly allowed since C++11